x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,165 @@
+package keeper
+
+import (
+	"sort"
+
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	evidencetypes "github.com/cosmos/cosmos-sdk/x/evidence/types"
+	ibcclienttypes "github.com/cosmos/ibc-go/v4/modules/core/02-client/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// HandleConsumerMisbehaviour checks if the given IBC misbehaviour corresponds to an equivocation light client attack,
+// and in this case, jails and tombstones the Byzantine validators
+func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) error {
+	logger := k.Logger(ctx)
+
+	// Check that the misbehaviour is valid and that the client isn't expired
+	if err := k.CheckMisbehaviour(ctx, misbehaviour); err != nil {
+		logger.Info("Misbehaviour rejected", err.Error())
+
+		return err
+	}
+
+	// Since the misbehaviour packet was received within the trusting period
+	// w.r.t to the last trusted consensus it entails that the infraction age
+	// isn't too old. see ibc-go/modules/light-clients/07-tendermint/types/misbehaviour_handle.go
+
+	// Get Byzantine validators from the conflicting headers
+	// Note that it returns an error if the misbehaviour doesn't correspond to an equivocation
+	byzantineValidators, err := k.GetByzantineValidators(ctx, misbehaviour)
+	if err != nil {
+		return err
+	}
+
+	// jail and tombstone the Byzantine validators
+	for _, v := range byzantineValidators {
+		// convert consumer consensus address
+		consuAddr := sdk.ConsAddress(v.Address.Bytes())
```

Comment:
We could convert this var to a `ConsumerConsAddress` directly and use the downcast method `ToSdkAddress()` as needed